promql/lex.go

```diff
@@ -86,6 +86,14 @@ func (i itemType) isSetOperator() bool {
 	return false
 }
 
+// isLabel returns whether the item can be used as label.
+func (i itemType) isLabel() bool {
+	if i == itemIdentifier || i.isOperator() || i.isAggregator() || i.isKeyword() {
```

Comment:
Not all operators can be used as label names, e.g. ==. Can you verify this doesn't allow this via unittest?